### PR TITLE
[Search] Enable test-proxy and test-resources.bicep

### LINKED ISSUE #20692: [Search] Test reliability issues with azure-search-documents
Two thorny issues that arose during the most recent Search work:

1. The `SearchServicePreparer` seemingly creates deterministic names instead of random ones. This causes tests to fail when the previous iteration of a test has not fully deleted. Like `StorageAccountPreparer`, the `SearchServicePreparer` should use a completely random name when running live that maps to a deterministic name to facilitate playback.
2. Sporadically, tests fail with a "Connection was force closed". The problem here is that this failure takes several minutes to manifest. The service should ideally _not_ fail, but if it does, it should fail fast.

### LINKED ISSUE #21493: [Search] Update Search to use Test Proxy server
https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/test_proxy_migration_guide.md

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #21493. Fixes #20692.

**Regular**
- [x] test_search_client_basic_live
- [x] test_search_client_buffered_sender_live
- [x] test_search_client_index_document_live
- [x] test_search_client_search_live
- [x] test_search_client_data_source_live
- [x] test_search_index_client_live
- [x] test_search_index_client_skillset_live
- [x] test_search_index_client_synonym_map_live
- [x] test_search_indexer_client_live

**Async**
- [x] test_search_client_basic_live_async
- [x] test_search_client_buffered_sender_live_async
- [x] test_search_client_index_document_live_async
- [x] test_search_client_search_live_async
- [x] test_search_client_data_source_live_async
- [x] test_search_index_client_live_async
- [x] test_search_index_client_skillset_live_async
- [x] test_search_index_client_synonym_map_live_asycn
- [x] test_search_indexer_client_live_async

### CODE REVIEW

**Comment by tjprescott** [comment]
Blocked by issue #22278.

**Comment by tjprescott** [comment]
Some tests had to be disabled because the service is failing for them, even on the main (unconverted) branch. See: https://github.com/Azure/azure-sdk-for-python/issues/22769

File: sdk/search/azure-search-documents/tests/test_search_indexer_client_live.py [hunk 1]
```diff
@@ -3,169 +3,173 @@
...
-
-class SearchIndexersClientTest(AzureMgmtTestCase):
-    FILTER_HEADERS = ReplayableTest.FILTER_HEADERS + ['api-key']
-
-    def _prepare_indexer(self, endpoint, api_key, name="sample-indexer", ds_name="sample-datasource", id_name="hotels"):
-        con_str = self.settings.AZURE_STORAGE_CONNECTION_STRING
-        self.scrubber.register_name_pair(con_str, 'connection_string')
-        container = SearchIndexerDataContainer(name='searchcontainer')
+
+class TestSearchIndexersClient(AzureRecordedTestCase):
```
Comment (tjprescott) [inline sdk/search/azure-search-documents/tests/test_search_indexer_client_live.py]: @mccoyp this is the one test I wrote (the first one) that uses variables. Compare it to the rest of the tests here.

**Comment by tjprescott** [comment]
@kashifkhan @mccoyp @xiangyan99 all tests have been converted and successfully record. This PR is ready for review!!

File: sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py [hunk 2]
```diff
@@ -3,79 +3,43 @@
...
             assert await client.get_document_count() == 10
 
-    @ResourceGroupPreparer(random_name_enabled=True)
-    @SearchServicePreparer(schema=SCHEMA, index_batch=BATCH)
-    async def test_get_document(self, api_key, endpoint, index_name, **kwargs):
-        client = SearchClient(
-            endpoint, index_name, AzureKeyCredential(api_key)
-        )
+    @SearchEnvVarPreparer()
+    @search_decorator(schema="hotel_schema.json", index_batch="hotel_small.json")
```
Comment (kashifkhan) [inline sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py:26]: curious, did Pytest have a warning around the test being async but not being decorated with the async marker due to strict mode being enabled in the future?

Comment (tjprescott) [reply sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py:26]: Nope, but some of the unit tests do complain about "async def function and no async plugin installed".

Comment (tjprescott) [reply sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py:26]: A quick pip install in my venv and those tests pass too--no warnings.

Comment (tjprescott) [reply sdk/search/azure-search-documents/tests/async_tests/test_search_client_basic_live_async.py:26]: image

**Review by kashifkhan — APPROVED** [review]
LGTM 👍

**Pull request merged**